src/fmu/dataio/export/rms/fluid_contact_surfaces.py

```diff
@@ -0,0 +1,179 @@
+        """Standard result type for the exported data."""
+        return standard_result.FluidContactSurfaceStandardResult(
+            name=StandardResultName.fluid_contact_surface
+        )
+
+    @property
+    def _content(self) -> Content:
+        """Get content for the exported data."""
+        return Content.fluid_contact
+
+    @property
+    def _classification(self) -> Classification:
+        """Get default classification."""
+        return Classification.internal
+
+    @property
+    def _rep_include(self) -> bool:
+        """rep_include status"""
+        return True
+
+    def _get_contacts(self, project: Any) -> list[FluidContactType]:
+        """
+        Get FluidContactTypes from available subfolder names in the main folder.
+        Folders with invalid contact names will be skipped. If no valid contact
+        names are found an error is raised.
+        """
+        contact_folders = list_folder_names_in_general2d_folder(
+            project, folder_path=[GENERAL2D_FOLDER]
+        )
+        valid_contact_folders = []
+        for contact in contact_folders:
+            try:
+                valid_contact_folders.append(FluidContactType(contact))
+            except KeyError:
+                _logger.info(f"{contact} is not a valid contact name, skipping folder.")
+                continue
+
+        _logger.debug(f"Found valid folders {valid_contact_folders}.")
+
+        if valid_contact_folders:
+            return valid_contact_folders
+
+        raise ValueError(
+            "Could not detect any fluid contact surfaces from RMS. "
+            f"Ensure the folder '{GENERAL2D_FOLDER}' exists in the "
+            "'General 2D data' folder, and that it contains minimum one subfolder "
+            f"with a valid contact name: {FluidContactType.__members__}. The contact "
+            "surfaces should be contained inside these subfolders."
+        )
+
+    def _get_contact_surfaces(
+        self, project: Any
+    ) -> dict[FluidContactType, list[xtgeo.RegularSurface]]:
+        """
+        Get a dictionary with fluid contact surfaces per contact folder
+        found in the main folder.
+        """
+        return {
+            contact: get_surfaces_in_general2d_folder(
+                project, folder_path=[GENERAL2D_FOLDER, contact.value]
+            )
+            for contact in self._get_contacts(project)
+        }
+
+    def _export_contact_surface(
+        self, contact: FluidContactType, surf: xtgeo.RegularSurface
+    ) -> ExportResultItem:
+        """Export a fluid contact surface as a standard result"""
+        edata = dio.ExportData(
+            config=self._config,
+            content=self._content,
+            content_metadata={"contact": contact, "truncated": False},
+            unit=self._unit,
+            vertical_domain="depth",
+            domain_reference="msl",
+            subfolder=f"{self._subfolder}/{contact.value}",
```

Comment:
I miss my `tagname` 😄  not sure how to best secure that data is not overridden for the different contacts, this does not feel like the most elegant solution but does the trick. Now the filename will be the surface name which is inline with other exports. 
Other option would be to fall back to the fmu standard of having attribute in the filename `surfname--attribute.gri` but today this is only possible through the `tagname` and we swore to keep that at `None.` So this felt like the best compromise at the moment. 
Any thoughts @mferrera ?